shared/src/balancer/event_handler.rs

```diff
@@ -155,7 +156,30 @@ impl PoolSpecialization {
     }
 }
 
-impl BalancerPools {
+impl BalancerPoolStore {
```

Comment:
nano-naming-nit (since you are already touching it). Maybe this is more a PoolRegistry?

Reply:
Should we also eliminate the term `Balancer` from this struct given that it is now in its own "balancer" module